### DISCUSSION TITLE
Container list flooded with containrrr/watchtower containers

### DISCUSSION THREAD

**Discussion by daniel-lo** [containrrr/watchtower#1600] [General] [answered]
Hi, my container list is flooded with containrrr/watchtower containers. I have no idea when this started but I did have former times with a clean container list. Clean meaning only those I installed/used.

Could anybody shed some light why they are there and how to potentially get rid of them (at least the 100 or so that make the list terribly long)?

Best

**Comment by piksel** [comment, marked as answer]
By, "container list", do you mean the output of `docker ps -a`?
If so you could filter the list by the label that watchtower uses to identify itself:
```
docker ps --filter label=com.centurylinklabs.watchtower=true --filter status=exited -a
```
and pass that to `docker rm`:
```
for c in $(docker ps --filter label=com.centurylinklabs.watchtower=true --filter status=exited -a -q); do docker rm $c; done
```

Not sure how you ended up with multiple instances though, but maybe it was caused by some bad release (it's surprisingly hard to not screw up watchtower releases due to it's self-updating nature). During normal operation watchtower should clean up any old instances of itself after updating.

**Comment by daniel-lo** [comment]
exactly!

docker ps --filter label=com.centurylinklabs.watchtower=true --filter status=exited -a

gives me an empty list though (I am not a programmer).

docker ps --filter label=com.centurylinklabs.watchtower=true --filter status=exited -a)

gives me a syntax error.

Best
